standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -10447,4 +10447,22 @@ public void drop_package(DropPackageRequest request) throws MetaException {
       endFunction("drop_package", ex == null, ex);
     }
   }
+
+  @Override
+  public List<WriteEventInfo> get_all_write_event_info(GetAllWriteEventInfoRequest request)
+      throws MetaException {
+    startFunction("get_all_write_event_info");
+    Exception ex = null;
+    try {
+      List<WriteEventInfo> writeEventInfoList =
+          getMS().getAllWriteEventInfo(request.getTxnId(), request.getDbName(), request.getTableName());
+      return writeEventInfoList == null ? Collections.emptyList() : writeEventInfoList;
+    } catch (Exception e) {
+      LOG.error("Caught exception", e);
+      ex = e;
```

Comment:
While existing method also has it, I wonder what do you get by this?

Reply:
`ex` is passed into `endFunction`. I guess the most obvious advantage is to get the function audited so that we can get some metrics (time, counter).